### PR TITLE
feat(portal-spec-renderer): create package

### PR DESCRIPTION
**Pull request by adamdehaven**
# Summary

Create `@kong-ui/portal-spec-renderer`

@kaiarrowood notice the package name is now different. We'll need to unpublish the old `@kong/portal-vue-spec-renderer` once this package is picked up.

## PR Checklist

* [ ] **Naming & Structure:** the files and package structure use the conventions outlined in the [Creating a Package docs](https://github.com/Kong/public-ui-components/blob/main/docs/creating-a-package.md).
* [ ] **Tests pass:** check the output of all package unit and/or component tests.
  * [ ] If this PR is the result of a bug, test coverage was added accordingly.
* [ ] **Functional:** all changes do not break existing APIs, but if so, a BREAKING CHANGE commit is in place to bump the major version.
* [ ] **Conventional Commits** all commits follow the conventional commit standards [outlined in the main README](https://github.com/Kong/public-ui-components#committing-changes).
* [ ] **Docs:** includes a technically accurate README, and the docs have been updated accordingly based on the changes in this PR.

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -28,6 +28,9 @@
 /packages/core/app-layout/ @adamdehaven @jillztom
 
 # Portal packages
+/packages/portal/document-viewer/ @kaiarrowood @mptap
+/packages/portal/vue-mini-spec-renderer/ @kaiarrowood @mptap
+/packages/portal/vue-spec-renderer/ @kaiarrowood @mptap
```
[the remark this answers is not in the document]
Comment (adamdehaven) [reply CODEOWNERS:33]: To be honest I'm not sure why @kaiarrowood prefixed the names with `vue`? 

This repository _**can and should**_ house any open-source, UI repositories. It is currently set up for Vue; however, we can update to support React if and should we (Kong) decide it necessary.

[the remark this answers is not in the document]
Comment (adamdehaven) [reply CODEOWNERS:33]: Correcting myself. I think I told Kai to name it this originally 😬  sorry @kaiarrowood 😂

[the remark this answers is not in the document]
Comment (mptap) [reply CODEOWNERS]: I'm in favor of removing the vue one just for simplicity and consistency with other packages

**Review by kaiarrowood — DISMISSED** [review]

File: CODEOWNERS (hunk 1 above)
[the remark this answers is not in the document]
Comment (kaiarrowood) [reply CODEOWNERS]: For historical purposes the final decision was to remove `vue` from the package name.

**Pull request merged**